src/pages/index.js

```diff
@@ -4,53 +4,136 @@ import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/Api.js";
 import "./index.css";
 import jacquesCousteau from "../images/jacques-cousteau.jpg";
 import logoImage from "../images/logo.svg";
 import {
   settings,
-  initialCards,
   profileEditButton,
   cardAddButton,
+  avatarEditButton,
 } from "../utils/constants.js";
 
 //Functions
 function handleEditProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profilePopupForm.close();
+  profilePopupForm.setLoading(true);
+  api
+    .editProfile({ name: data.name, description: data.description })
+    .then((res) => {
+      userInfo.setUserInfo({ name: res.name, description: res.about });
+      profilePopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  profilePopupForm.setLoading(false);
   return;
 }
 
+function handleAvatarFormSubmit(data) {
+  avatarPopupForm.setLoading(true);
+  api
+    .updateAvatar(data.image)
+    .then((res) => {
+      avatarPopupForm.close();
+      avatarPopupForm.reset();
+      userInfo.setAvatar(data.image);
+    })
+    .catch((err) => {
+      console.error("Error saving avatar:", err);
+    })
+    .finally(avatarPopupForm.setLoading(false));
+}
+
 function handleAddProfileFormSubmit(data) {
-  const name = data.name;
-  const link = data.image;
-  const cardElement = createCard({ name, link });
-  layerSection.addItem(cardElement);
-  addCardPopupForm.close();
-  addCardPopupForm.reset();
-  addFormValidator.toggleButtonState();
+  addCardPopupForm.setLoading(true);
+  api
+    .addCard({
+      name: data.name,
+      link: data.image,
+    })
+    .then((res) => {
+      layerSection.addItem(createCard(res));
+      addCardPopupForm.close();
+      addCardPopupForm.reset();
+      addFormValidator.toggleButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  addCardPopupForm.setLoading(false);
   return;
 }
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleCardImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleCardImageClick,
+    handleCardDelete,
+    handleLikeCard,
+    handleDeleteModal
+  );
   return card.renderCard();
 }
 
 function handleCardImageClick(name, link) {
   cardImagePopup.open({ name, link });
 }
 
+function handleCardDelete(card) {
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      card._cardElement.remove();
```

Comment:
`_cardElement`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.


**Please, correct it everywhere in the code of `index.js`**

Reply:
Thank you! Fixed.

Reply:
Instances found: _cardElement, _id, _isLiked --> All fixed.